Description: Solve syntax issue when using libglib2 >= 2.67
  Preprocessor output for `::g_object_ref(gicon); ` is:
      ::((typename std::remove_reference<decltype (gicon)>::type) (g_object_ref) (gicon));
  Since commit https://gitlab.gnome.org/GNOME/glib/-/commit/5b2bee3f539056b42c802608f9f00cc9ddd64b79
  that is shipped starting from 2.67.0, the macro g_object_ref changed and is
  also defined when __cplusplus is defined in some circumstances.
 .
  Removing the scope resolution operator `::` permits the code to build.
  There remains to check whether this is the correct fix.
 .
 See also: https://freefilesync.org/forum/viewtopic.php?t=8780
 .
 Required since 11.21
 .
Last-Update: 2022-08-08
Version: 11.21
Author: bgstack15
Author: Fab Stz <fabstz-it@yahoo.fr>
Forwarded: https://freefilesync.org/forum/viewtopic.php?t=8780#p34828

--- a/FreeFileSync/Source/base/icon_loader.cpp
+++ b/FreeFileSync/Source/base/icon_loader.cpp
@@ -250,7 +250,11 @@
     //the remaining icon types won't block!
     assert(GDK_IS_PIXBUF(gicon) || G_IS_THEMED_ICON(gicon) || G_IS_EMBLEMED_ICON(gicon));
 
+#if (GLIB_CHECK_VERSION (2, 67, 0))
+    g_object_ref(gicon);                   //pass ownership
+#else
     ::g_object_ref(gicon);                 //pass ownership
+#endif
     return FileIconHolder(gicon, maxSize); //
 
 }
